controller/volume_controller.go

```diff
@@ -2887,8 +2875,11 @@ func (vc *VolumeController) processMigration(v *longhorn.Volume, es map[string]*
 
 	log := getLoggerForVolume(vc.logger, v).WithField("migrationNodeID", v.Spec.MigrationNodeID)
 
-	// only process if volume is attached
-	if v.Spec.NodeID == "" || len(es) == 0 {
+	// only process if volume is attached and running
+	if v.Spec.NodeID == "" || v.Status.CurrentNodeID == "" || len(es) == 0 {
+		return nil
+	}
+	if v.Status.Robustness != longhorn.VolumeRobustnessDegraded && v.Status.Robustness != longhorn.VolumeRobustnessHealthy {
```

Comment:
So we only process the migration when the volume is not degraded? Should we add some debug log to better know why the volume can't be migrated here?

Reply:
We only process the migration when the volume is **neither** degraded nor **healthy**. I am afraid that that kind of message will flood the log.

Reply:
Maybe tracing log? I know we use debug by default.

Reply:
The trace log actually does not make too much sense... Do we really need that?